### PR TITLE
persist: repurpose open loop benchmark for platform

### PR DESCRIPTION
**Pull request by ruchirK**
This commit brings back the open loop benchmark with some limitations for use
with new versions of materialized/computed intended for platform.

The open loop benchmark:
- no longer supports the upsert envelope.
- currently only supports a single computed process with a single worker.
- currently only supports a single storaged process.

### Testing

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.

### Release notes

This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  -

### CODE REVIEW

**Comment by ruchirK** [comment]
running with s3 + expensive aurora pg on a persist benchmarking machine I'm able to have this do 200k requests / second with 8 byte key / 500 byte values per request. At that throughput materialize exhibits max 20 seconds worth of lag.

To run:

```
./mzcompose run default --consensus-url postgres://postgres:danaljoscha@database-2.cluster-cbjgjaqxcurd.us-east-2.rds.amazonaws.com --blob-url s3://mtlz-test-persist-1d-lifecycle-delete/ruchir-testing-e2e-2/ --records-per-second 200000
```

and the results:

```
C> after 1.631s sent 37245 records, and ingested 0. max observed lag 37245 records, most recent lag 37245 records
C> after 4.279s sent 326199 records, and ingested 0. max observed lag 326199 records, most recent lag 326199 records
C> after 7.428s sent 855872 records, and ingested 37245. max observed lag 818627 records, most recent lag 818627 records
C> after 11.446s sent 1485617 records, and ingested 640266. max observed lag 845351 records, most recent lag 845351 records
C> after 16.204s sent 2289239 records, and ingested 1485617. max observed lag 845351 records, most recent lag 803622 records
C> after 21.545s sent 3240806 records, and ingested 2171740. max observed lag 1069066 records, most recent lag 1069066 records
C> after 27.710s sent 4308905 records, and ingested 3240807. max observed lag 1069066 records, most recent lag 1068098 records
C> after 34.208s sent 5541909 records, and ingested 4308905. max observed lag 1233004 records, most recent lag 1233004 records
C> after 40.391s sent 6841575 records, and ingested 5069838. max observed lag 1771737 records, most recent lag 1771737 records
C> after 50.370s sent 8078162 records, and ingested 5541909. max observed lag 2536253 records, most recent lag 2536253 records
C> after 62.589s sent 10074084 records, and ingested 8330986. max observed lag 2536253 records, most recent lag 1743098 records
C> after 74.411s sent 12517840 records, and ingested 10315575. max observed lag 2536253 records, most recent lag 2202265 records
C> after 87.124s sent 14882104 records, and ingested 12866987. max observed lag 2536253 records, most recent lag 2015117 records
C> after 98.228s sent 17424720 records, and ingested 14166448. max observed lag 3258272 records, most recent lag 3258272 records
C> after 109.140s sent 19645687 records, and ingested 15321897. max observed lag 4323790 records, most recent lag 4323790 records
C> after 111.980s sent 20000000 records, and ingested 15321897. max observed lag 4678103 records, most recent lag 4678103 records
C> after 113.713s sent 20000000 records, and ingested 16086407. max observed lag 4678103 records, most recent lag 3913593 records
C> after 115.685s sent 20000000 records, and ingested 16599568. max observed lag 4678103 records, most recent lag 3400432 records
C> after 118.769s sent 20000000 records, and ingested 17407335. max observed lag 4678103 records, most recent lag 2592665 records
C> after 120.061s sent 20000000 records, and ingested 17811085. max observed lag 4678103 records, most recent lag 2188915 records
C> after 121.686s sent 20000000 records, and ingested 18214637. max observed lag 4678103 records, most recent lag 1785363 records
C> after 128.233s sent 20000000 records, and ingested 19194971. max observed lag 4678103 records, most recent lag 805029 records
C> after 129.234s sent 20000000 records, and ingested 19194971. max observed lag 4678103 records, most recent lag 805029 records
C> after 130.399s sent 20000000 records, and ingested 19194971. max observed lag 4678103 records, most recent lag 805029 records
C> after 131.717s sent 20000000 records, and ingested 19221354. max observed lag 4678103 records, most recent lag 778646 records
C> after 137.846s sent 20000000 records, and ingested 19221359. max observed lag 4678103 records, most recent lag 778641 records
C> after 140.729s sent 20000000 records, and ingested 19869791. max observed lag 4678103 records, most recent lag 130209 records
C> Finished after 141.242s sent and ingested 20000000 records. max observed lag 4678103 records.
```

File: test/kafka-ingest-open-loop/mzcompose.py [hunk 1]
```diff
@@ -236,13 +196,10 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
...
-        )
-        c.kill("materialized")
-        c.rm("materialized", "testdrive", "kafka", destroy_volumes=True)
-        c.rm_volumes("mzdata", "pgdata")
+    c.workflow(
+        "default",
+        "--num-seconds=15",
+        "--records-per-second=1000",
+    )
+    c.rm("materialized", "testdrive", "kafka", destroy_volumes=True)
```
Comment (danhhz) [inline test/kafka-ingest-open-loop/mzcompose.py:204]: did we intentionally lose the `c.kill("materialized")` that was here before?

**Review by danhhz — APPROVED** [review]
LGTM!

**Comment by aljoscha** [comment]
Could you remind me what the numbers where for the previous incarnation?

File: test/kafka-ingest-open-loop/mzcompose.py (hunk 1 above)
Comment (ruchirK) [reply test/kafka-ingest-open-loop/mzcompose.py:205]: unintentional. fixed!

**Comment by ruchirK** [comment]
TFTRS!!

**Pull request merged**